lib/utils/matrix_sdk_extensions/event_extension.dart

```diff
@@ -158,4 +165,66 @@ extension LocalizedBody on Event {
       room.isDirectChat ||
       !isSameSenderWith(nextEvent) ||
       type == EventTypes.Encrypted;
+
+  bool get isPinned => room.pinnedEventIds.contains(eventId);
+
+  Future<bool> unpin() async {
+    if (!isPinned) {
+      return true;
+    }
+    final pinnedEventIds = room.pinnedEventIds;
+    pinnedEventIds.remove(eventId);
+    String? message;
+    try {
+      message = await room.setPinnedEvents(pinnedEventIds);
+    } catch (e) {
+      Logs().e('$e');
+      return false;
+    }
+    if (message.startsWith('\$')) {
+      return true;
+    }
+    return false;
```

Comment:
Why must it be handled like this?

Reply:
Because the API from matrix sdk return a string, and the eventId always start with $, this is the only way. Otherwise we have to write the API again
![image](https://github.com/linagora/twake-on-matrix/assets/43041967/2fbaa9ba-15be-4cea-bd4a-b84f3497cb08)